### PR TITLE
Improve systemd hardening

### PR DESCRIPTION
**Pull request by alexAubin**
## Changes

- A couple minor things for systemd hardening taken from https://news.ycombinator.com/item?id=29976096. Not a huge deal, i.e. it's not like we gotta absolutely apply this to every app already implementing some form of hardening, it's just nice to have.

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by tituspijean** [comment]
Oooh, what about this one? Seems to be a perfect fit for most apps serving only on an internal port.

```
  SocketBindDeny=any
  SocketBindAllow=tcp:__PORT__
```

File: conf/systemd.service [hunk 1]
```diff
@@ -18,16 +18,21 @@ StandardError=inherit
...
 PrivateDevices=yes
-RestrictAddressFamilies=AF_UNIX AF_INET AF_INET6
+RestrictAddressFamilies=AF_UNIX AF_INET AF_INET6 AF_NETLINK
 RestrictNamespaces=yes
 RestrictRealtime=yes
 DevicePolicy=closed
+ProtectClock=yes
+ProtectHostname=yes
+ProtectKernelLogs=yes
+ProtectProc=invisible
```
Comment (tituspijean) [inline conf/systemd.service:28]: ⚠️ These unit file settings were not introduced within the same Systemd version and will not all work.
My current YunoHost v4.3.6 ships Systemd version 241, for example. A recently installed beta YunoHost 11.0.2 ships Systemd 247.3-6.

| unit file settings | Systemd version |
|------------------|-----------------| 
`ProtectClock` | 247
`ProtectHostname` | 242
`ProtectKernelLogs` | 250
`ProtectProc` | 247

Comment (alexAubin) [reply conf/systemd.service:28]: Ah, thanks for checking this :sweat_smile: 

So not merging before bullseye, and even then we shall only have `ProtectClock`, `Hostname` and `Proc` ?

**Review by ericgaspar — APPROVED** [review]

File: conf/systemd.service (hunk 1 above)
Comment (Tagadda) [reply conf/systemd.service:28]:
```suggestion
ProtectClock=yes
ProtectHostname=yes
ProtectProc=invisible
```

**Review by Tagadda — CHANGES_REQUESTED** [review]

**Pull request merged**